tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -510,3 +511,46 @@ def vdc_allocate_interface_set(vdc, intf)
   cmd = get_vshell_cmd('no terminal dont-ask persist')
   on(agent, cmd, pty: true)
 end
+
+# Facter command builder helper method
+def facter_cmd(cmd)
+  get_namespace_cmd(agent, FACTER_BINPATH + cmd, options)
+end
+
+# Used to cache the operation system information
+@cisco_os = nil
+# Use facter to return cisco operating system information
+def operating_system
+  return @cisco_os unless @cisco_os.nil?
+  @cisco_os = on(agent, facter_cmd('os.name')).stdout.chomp
```

Comment:
It's a classic ruby construct. What's not to love? :-)
Okay, not a blocker, I'll merge it.